core/arch/x86/encode.c

```diff
@@ -746,6 +759,11 @@ size_ok(decode_info_t *di /*prefixes field is IN/OUT; x86_mode is IN*/,
         case OPSZ_16:
             if (X64_MODE(di) &&
                 (size_template == OPSZ_8_rex16 || size_template == OPSZ_8_rex16_short4)) {
+                /* XXX #3581: when encoding and the operand is a TYPE_T_REG or
+                 * TYPE_T_MODRM register, this causes a rex prefix. The rex prefix is
+                 * ignored for the MPX instruction that are affected, but it is not
+                 * necessary.
```

Comment:
I'm confused: why would this happen when the TYPE_T* entries in the table never have this size?

Reply:
This comment is stale. Removed. It doesn't happen anymore after I changed it to using OPSZ_8_dimode16 instead of OPSZ_8_rex16. I will change into OPSZ_8x16.

Reply:
Done.